### ISSUE TITLE
getting representative docs after online-fitting

### ISSUE THREAD

**Issue by maciejskorski** [MaartenGr/BERTopic#1679]
Often `partial_fit` is invoked multiple times to process a large dataset. 

It seems that the attribute `representative_docs_` is not populated then. 
Is there an easy way to get representative docs in such a case?

**Comment by maciejskorski** [comment]
@MaartenGr thanks! Following your suggestion, here is a snippet **using `_create_topic_vectors` and `_save_representative_docs` internal functions**. 

Suppose that `docs` are documents, `embeds` are their embeddings, `topic_model` is the model fitted online and `train_idxs` are the indexes in the shuffled order (if applicable). We **populate topic representations first**, then we are in a position to **populate representative documents next**:

```python
doc_topic = pd.DataFrame({'Topic':topic_model.topics_,'ID':range(len(topic_model.topics_)),'Document':docs.loc[train_idxs]})
topic_model._create_topic_vectors(doc_topic,embeds[train_idxs])
#topic_model._save_representative_docs(doc_topic)
repr_docs, _, _, _=  topic_model._extract_representative_docs(
    topic_model.c_tf_idf_, 
    doc_topic,
    topic_model.topic_representations_,
    nr_samples=1000,
    nr_repr_docs=5
)
topic_model.representative_docs_ = repr_docs
```
I tested this on >1M documents, here is an example:
<img width="1616" alt="image" src="https://github.com/MaartenGr/BERTopic/assets/31315784/93dadba8-5bfd-4f71-a3da-2a670a91ba9a">

**Comment by MaartenGr** [comment]
That's great, thanks for sharing! Other users will definitely benefit from having this code snippet here.

**Comment by maciejskorski** [comment]
> That's great, thanks for sharing! Other users will definitely benefit from having this code snippet here.

@MaartenGr if you don't mind, I would volunteer to raise a PR expanding a bit [the online tutorial example](https://github.com/MaartenGr/BERTopic/blob/master/docs/getting_started/online/online.md), demonstrating the use of these internal functions on News20?

**Comment by linxule** [comment]
[like]  Lin, Xule reacted to your message:
________________________________

**Comment by linxule** [comment]
[0]     Lin, Xule reacted to your message:
________________________________

**Comment by keyuchen21** [comment]
@maciejskorski  will that be possible to show a complete example other than the small snippet?
1) i'm a little bit confused by what's your "train_idxs"?
2) could you help me take a look of where I got it wrong?

thank you for your help!!!!

from bertopic import BERTopic
from bertopic.representation import KeyBERTInspired
from bertopic.representation import PartOfSpeech
from bertopic.representation import MaximalMarginalRelevance
from sklearn.datasets import fetch_20newsgroups

# Documents to train on
docs = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))['data'][:500]

# The main representation of a topic
main_representation = KeyBERTInspired()

# Additional ways of representing a topic
aspect_model1 = PartOfSpeech("en_core_web_sm")
aspect_model2 = [KeyBERTInspired(top_n_words=30), MaximalMarginalRelevance(diversity=.5)]

# Add all models together to be run in a single `fit`
representation_model = {
   "Main": main_representation,
   "Aspect1":  aspect_model1,
   "Aspect2":  aspect_model2 
}
topic_model = BERTopic(representation_model=representation_model).fit(docs)
doc_topic = topic_model.get_topic_info()

repr_docs, _, _, _=  topic_model._extract_representative_docs(
    topic_model.c_tf_idf_, 
    doc_topic,
    topic_model.topic_representations_,
    nr_samples=1000,
    nr_repr_docs=5
)

<img width="1167" alt="Image" src="https://github.com/user-attachments/assets/f90844b3-c395-40ce-b0ae-7efdd22266ec" />

**Comment by keyuchen21** [comment]
I found this example to be useful, in case anyone need it:

```
from bertopic import BERTopic
from bertopic.representation import KeyBERTInspired
from bertopic.representation import PartOfSpeech
from bertopic.representation import MaximalMarginalRelevance
from sklearn.datasets import fetch_20newsgroups

# Documents to train on
docs = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))['data'][:500]

# The main representation of a topic
main_representation = KeyBERTInspired()

# Additional ways of representing a topic
aspect_model1 = PartOfSpeech("en_core_web_sm")
aspect_model2 = [KeyBERTInspired(top_n_words=30), MaximalMarginalRelevance(diversity=.5)]

# Add all models together to be run in a single `fit`
representation_model = {
   "Main": main_representation,
   "Aspect1":  aspect_model1,
   "Aspect2":  aspect_model2 
}
topic_model = BERTopic(representation_model=representation_model).fit(docs)


import pandas as pd

# Prepare your documents to be used in a dataframe
documents = pd.DataFrame({"Document": docs,
                          "ID": range(len(docs)),
                          "Topic": topic_model.topics_})

# Extract the top 50 representative documents
repr_docs, _, _, _= topic_model._extract_representative_docs(c_tf_idf=topic_model.c_tf_idf_,
                                                          documents=documents,
                                                          topics=topic_model.topic_representations_ ,
                                                          nr_repr_docs=50)
```
